src/aerovaldb/jsondb/cache.py

```diff
@@ -53,39 +57,88 @@ def remove(self, item: Hashable):
             self._deque.remove(item)
 
 
+class BaseCache(ABC):
+    def __init__(self):
+        pass
+
+    @property
+    @abstractmethod
+    def hit_count(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def size(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def miss_count(self) -> int:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get(self, key: str, *, bypass_cache: bool = False) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def put(self, obj: str, *, key: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def invalidate_all(self) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def invalidate_entry(self, file_path: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_valid(self, key: str) -> bool:
+        raise NotImplementedError
+
+
 class CacheEntry(TypedDict):
     json: str
 
     last_modified: float
 
 
-class JSONLRUCache:
+class LRUFileCache(BaseCache):
```

Comment:
This sounds very general, but why do you have the functions `get_json_entry`... which seems to be very specific, the old name and explanation was much more specific.